File: academy/academy_rest_api/serializers/exercises.py
from rest_framework import serializers

from exercises.models import Exercise


class ExerciseSerializer(serializers.HyperlinkedModelSerializer):
    class Meta:
        model = Exercise
        fields = ['exercise_id', 'name', 'description', 'tags', 'status']



# code format serializer
class CodeFormatSerializer(serializers.Serializer):
    code = serializers.CharField()

# code analysis serializer
class CodeAnalysisSerializer(serializers.Serializer):
    code = serializers.CharField(required=True)
    disable_errors = serializers.ListField(
        child=serializers.CharField(), required=False, default=[]
    )

File: react_frontend/src/types/exercises.ts
export interface Exercise {
  exercise_id: string;
  name: string;
  description: string;
  tags: string;
  status: ExerciseStatus;
}

export interface ExerciseData {
  name: string;
  exercise_id: string;
  url: string;
  tools: string[];
  tags: string[];
  worlds: object[];
}

export type ExerciseStatus = "ACTIVE" | "INACTIVE" | "PROTOTYPE";
export type Filters = "name" | "tags" | "description" | "status";
